File: calculator.py
#calculator.py functions start here:
# class calculator:
#add()
def add(x, y):
    return x + y

#subtract()
def subtract(x, y):
    return x - y

#multiply()
def multiply(x, y):
    return x * y

#divide()
def divide(x, y):
    if y == 0:
        return "Error: Division by zero!"
    return x / y

File: 112-final-a/Calculator.java
public class Calculator {
  private double result;

  public Calculator() {
    result = 0.0;
  }

  public void addition(double val) {
    result += val;
  }

  public void subtraction(double val) {
    result -= val;
  }

  public void multiplication(double val) {
    result *= val;
  }

  public void division(double val) {
    if (val != 0) {
      result /= val;
    } else {
      System.out.println("Error: Division by zero");
    }
  }

  public void clear() {
    result = 0.0;
  }

  public double getValue() {
    return result;
  }
}